bots/tests-dequeue-one

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+MAX_PRIORITY = 12
+
+import argparse
+import pika
+import subprocess
+import sys
+
+def main():
+    parser = argparse.ArgumentParser(description='Bot: read test commands from the queue and execute them')
+    parser.add_argument('--amqp', action="store_true", default='localhost',
+                        help='The url of the amqp server to consume from.')
+    parser.add_argument('--queue', action="store_true", default='tests',
+                        help='The name of the queue to consume from.')
+    opts = parser.parse_args()
+
+    connection = pika.BlockingConnection(pika.ConnectionParameters(host=opts.amqp))
+    channel = connection.channel()
+    arguments = {
+        'x-max-priority': MAX_PRIORITY
+    }
+    channel.queue_declare(queue=opts.queue, arguments=arguments)
+    channel.basic_qos(0, 1, True)
+    channel.basic_consume(run_test, queue=opts.queue)
+    channel.start_consuming()
+
+def run_test(ch, method, properties, body):
+    command = body.decode('utf-8')
+
+    try:
+        # exists zero on test completion (so not just on positive test result)
+        subprocess.check_call(command.split(' '))
```

Comment:
I don't think this works. First of all, tests-scan produces two commands, not one:

    
PRIORITY=0006 bots/make-checkout --verbose --base=master --repo=weldr/welder-web --bots-ref=pull/10716/head '' && TEST_PROJECT=weldr/welder-web TEST_NAME=pull-10716-20181130-170707 TEST_REVISION=4cd21429e3cb4fe7db97e518858f6f69106b0254 bots/tests-invoke --rebase=master --remote=test cockpit/fedora-29/chrome ''

And second, if you have quoted arguments that contain spaces, this simple-minded split would break them apart. I think it's better to just use `shell=True`.